File: store/src/main/java/com/min/store/product/entity/ProductEntity.java
package com.min.store.product.entity;

import com.min.store.product.domain.ProductDomain;
import lombok.*;

import javax.persistence.*;

@Getter
@Table(name = "PRODUCT")
@Entity
public class ProductEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private Long price;

    public static ProductEntity from(ProductDomain productDomain) {
        ProductEntity productEntity = new ProductEntity();
        productEntity.id = productDomain.getId();
        productEntity.name = productDomain.getName();
        productEntity.price = productDomain.getPrice();
        return productEntity;
    }

    public ProductDomain toDomain() {
        return ProductDomain.builder()
                .id(id)
                .name(name)
                .price(price)
                .build();
    }

}

File: store/store-product-api/src/main/kotlin/com/store/product/api/entity/ProductEntity.kt
package com.store.product.api.entity

import com.store.product.api.domain.ProductDomain
import javax.persistence.*

@Table(name = "PRODUCT")
@Entity
class ProductEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,
    val name: String,
    val price: Long,
) {

    companion object {
        fun from(productDomain: ProductDomain): ProductEntity {
            return ProductEntity(
                id= productDomain.id,
                name = productDomain.name,
                price = productDomain.price
            )
        }
    }

    fun toDomain() : ProductDomain {
        return ProductDomain(id = id, name = name, price = price)
    }
}
